config/routes.rb

```diff
@@ -1,6 +1,14 @@
 Rails.application.routes.draw do
+
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  get '/users', to: 'users#index', as: 'users' 
+  get '/users/:id', to: 'users#show', as: 'user'
+
+  root 'users#index'
```

Comment:
- [ ] [OPTIONAL] Kindly write the routes in this way:
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```
as it's cleaner and easier to read and refactor.